app/src/main/res/layout/fragment_home.xml

```diff
@@ -167,11 +174,38 @@
                             app:layout_constraintTop_toBottomOf="@+id/layout_home_exposure_labels"
                             app:layout_constraintEnd_toEndOf="parent"
                             app:layout_constraintStart_toStartOf="parent"
-                            android:visibility="@{model.hasExposures==true ? View.VISIBLE : View.GONE}"
+                            android:visibility="@{model.hasExposures()==true ? View.VISIBLE : View.GONE}"
                             android:text="@string/home_exposure_instructions"
                             android:theme="@style/ThemeOverlay.Vilkku.Button.Critical"
                             tools:visibility="gone"/>
 
+                        <com.google.android.material.button.MaterialButton
+                            android:id="@+id/button_home_exposure_check"
+                            android:layout_width="0dp"
+                            android:layout_height="wrap_content"
+                            android:layout_marginTop="20dp"
+                            app:layout_constraintTop_toBottomOf="@+id/layout_home_exposure_labels"
+                            app:layout_constraintEnd_toEndOf="parent"
+                            app:layout_constraintStart_toStartOf="parent"
+                            android:visibility="@{model.showManualCheck()==true ? View.VISIBLE : View.GONE}"
+                            android:text="@string/home_exposure_check"
+                            android:enabled="@{!model.checkInProgress}"
+                            tools:visibility="visible"
+                            tools:enabled="false"
+                            />
+
+                        <ProgressBar
+                            android:layout_width="26dp"
```

Comment:
Progress bar / circle could be a bit smaller imho (20dp + marginStart→10 or 12dp) - this way it would "line up" with the radius of the button (with the default text size) :)

Reply:
thanks, looks better with 20dp